netdata-installer.sh

```diff
@@ -40,6 +40,24 @@ fi
 # make sure /etc/profile does not change our current directory
 cd "${NETDATA_SOURCE_DIR}" || exit 1
 
+# -----------------------------------------------------------------------------
+# set up handling for deferred error messages
+NETDATA_DEFERRED_ERRORS=""
+
+defer_error() {
+  NETDATA_DEFERRED_ERRORS="${NETDATA_DEFERRED_ERRORS}\n* ${1}"
+}
+
+print_deferred_errors() {
+  if [ -n "${NETDATA_DEFERRED_ERRORS}" ] ; then
+    echo >&2
+    echo >&2 "The following non-fatal errors were encountered during the installation process:"
+    # shellcheck disable=SC2059
+    printf >&2 "${NETDATA_DEFERRED_ERRORS}"
```

Comment:
```suggestion
    printf >&2 "%s" "${NETDATA_DEFERRED_ERRORS}"
```

Reply:
I'm not 100% certain that this achieves what my intent was. I'm trying to deal with the fact that `echo` did not, for some reason, expand `\n` properly in the variable, which in and of itself didn't have it expanded properly for reasons I don't really understand.

Reply:
echo by default will not expand or interpret blackslash sequences Unless you use the `-e` option:

> -e     enable interpretation of backslash escapes

Reply:
That's only some versions of `echo`, not all support it (FreeBSD's implementation in particular does not).